query/src/main/scala/filodb/query/exec/rangefn/RateFunctions.scala

```diff
@@ -22,7 +22,7 @@ object RateFunctions {
       var durationToStart = (window.head.timestamp - startTimestamp).toDouble / 1000
       val durationToEnd = (endTimestamp - window.last.timestamp).toDouble / 1000
       val sampledInterval = (window.last.timestamp - window.head.timestamp).toDouble / 1000
-      val averageDurationBetweenSamples = sampledInterval / window.size.toDouble
+      val averageDurationBetweenSamples = sampledInterval / (window.size.toDouble - 1)
```

Comment:
Is there a unit test to verify this?

Reply:
Is it possible for the window size to be 1, causing a DivideByZero to be thrown?

Reply:
@velvia as a double, it will end up being infinity or NaN.

Reply:
@vishramachandran @broneill @velvia added a test case, which was failing earlier without the fix. RawData/ResultSet is taken from prometheus engine